### PR TITLE
Fix Common AuthSecret Name

### PR DESCRIPTION
**Pull request by raihankhan**

### CODE REVIEW

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]
kafka ? 
```
arnob@office ~> kubectl get secrets -n demo | grep kafka-
kafka-test-admin-cred                 kubernetes.io/basic-auth   2      5h21m
kafka-test-config                     Opaque                     2      5h21m
```

**Comment by obaydullahmhs** [comment]
Already opened a PR: https://github.com/kubedb/apimachinery/pull/1331

File: apis/kubedb/v1alpha2/druid_helpers.go [hunk 1]
```diff
@@ -577,18 +583,6 @@ func (d *Druid) SetDefaults() {
...
-		d.SetTLSDefaults()
-	}
-}
-
-func (d *Druid) SetTLSDefaults() {
-	if d.Spec.TLS == nil || d.Spec.TLS.IssuerRef == nil {
-		return
-	}
-	d.Spec.TLS.Certificates = kmapi.SetMissingSecretNameForCertificate(d.Spec.TLS.Certificates, string(DruidServerCert), d.CertificateName(DruidServerCert))
-	d.Spec.TLS.Certificates = kmapi.SetMissingSecretNameForCertificate(d.Spec.TLS.Certificates, string(DruidClientCert), d.CertificateName(DruidClientCert))
```
Comment (tapojit047) [inline apis/kubedb/v1alpha2/druid_helpers.go:591]: These lines should not be removed

File: apis/kubedb/v1alpha2/druid_helpers.go [hunk 2]
```diff
@@ -791,16 +785,3 @@ func (d *Druid) GetZooKeeperName() string {
...
-func (d *Druid) CertSecretVolumeName(alias DruidCertificateAlias) string {
-	return string(alias) + "-certs"
-}
-
-// CertSecretVolumeMountPath returns the CertSecretVolumeMountPath
-// if configDir is "/var/druid/ssl",
-// mountPath will be, "/var/druid/ssl/<alias>".
-func (d *Druid) CertSecretVolumeMountPath(configDir string, cert string) string {
-	return filepath.Join(configDir, cert)
-}
```
Comment (tapojit047) [inline apis/kubedb/v1alpha2/druid_helpers.go:806]: These lines should not be removed as well

**Review by tapojit047 — CHANGES_REQUESTED** [review]

**Review by pritamdas99 — APPROVED** [review]
LGTM!

**Review by sheikh-arman — COMMENTED** [review]
lgtm

**Review by tapojit047 — DISMISSED** [review]
LGTM!

File: apis/kubedb/v1alpha2/solr_helpers.go [hunk 3]
```diff
@@ -262,12 +269,6 @@ func (s *Solr) SetDefaults() {
 		s.Spec.StorageType = StorageTypeDurable
 	}
 
-	if s.Spec.AuthSecret == nil {
-		s.Spec.AuthSecret = &v1.LocalObjectReference{
-			Name: s.SolrSecretName("admin-cred"),
-		}
```
Comment (ArnobKumarSaha) [inline apis/kubedb/v1alpha2/solr_helpers.go:268]: We dont need the authSecret defaulting ?

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]

File: apis/kubedb/v1alpha2/solr_helpers.go (hunk 3 above)
Comment (pritamdas99) [reply apis/kubedb/v1alpha2/solr_helpers.go:268]: no
a method is added to retrieve the name of auth secret.

**Pull request merged**